hack/generator/pkg/astmodel/resource.go

```diff
@@ -81,12 +83,25 @@ func NewAzureResourceType(specType Type, statusType Type, typeName TypeName) *Re
 			isNameOptional = true
 		}
 
+		if apiVersionProperty == nil {
+			panic(fmt.Sprintf("Resource %s is missing apiVersion property", typeName))
+		}
+
 		if isNameOptional {
 			// Fix name to be required -- again this is an artifact of bad spec more than anything
 			nameProperty = nameProperty.MakeRequired()
 			objectType = objectType.WithProperty(nameProperty)
 		}
 
+		// Fix APIVersion to be optional. Technically this isn't due to a bad specification, but in our
+		// case forcing it to required makes our lives simpler (and the vast majority of resources specify
+		// it as required anyway). The only time it's allowed to be optional is if you set apiProfile on
+		// the ARM template instead, which we never do.
+		if apiVersionOptional {
```

Comment:
If we always want `apiVersion` to be required it would be clearer to do it unconditionally, rather than checking to see if it's optional and only updating it when it is.

Reply:
Yes, yes it would.

Will fix!